pulp_container/app/serializers.py

```diff
@@ -293,6 +296,15 @@ def validate(self, data):
         """
         new_data = super().validate(data)
 
+        if not re.match(VALID_TAG_REGEX, data["tag"]):
+            raise serializers.ValidationError(
+                _(
+                    "The provided tag '{}' is not valid. A tag may contain lowercase and uppercase "
+                    "ASCII alphabetic characters, digits, underscores, periods, and dashes. A tag "
+                    "must not start with a period or a dash.".format(data["tag"])
+                )
```

Comment:
```suggestion
                _(
                    "The provided tag '{}' is not valid. A tag may contain lowercase and uppercase "
                    "ASCII alphabetic characters, digits, underscores, periods, and dashes. A tag "
                    "must not start with a period or a dash."
                ).format(data["tag"])
```

Reply:
Do you think, https://www.django-rest-framework.org/api-guide/fields/#regexfield helps here?